be/src/storage/lake/tablet.cpp

```diff
@@ -58,8 +58,26 @@ StatusOr<std::unique_ptr<TabletWriter>> Tablet::new_writer() {
 }
 
 StatusOr<std::shared_ptr<const TabletSchema>> Tablet::get_schema() {
-    ASSIGN_OR_RETURN(auto metadata, get_metadata(1));
-    return GlobalTabletSchemaMap::Instance()->emplace(metadata->schema()).first;
+    auto tablet_schema_key = _mgr->tablet_schema_cache_key(_id);
+    auto ptr = _mgr->lookup_tablet_schema(tablet_schema_key);
+    RETURN_IF(ptr != nullptr, ptr);
+
+    ASSIGN_OR_RETURN(TabletMetadataIter metadata_iter, _mgr->list_tablet_metadata(_group, _id));
+    if (!metadata_iter.has_next()) {
+        return Status::NotFound(fmt::format("tablet {} metadata not found", _id));
+    }
+    ASSIGN_OR_RETURN(auto metadata, metadata_iter.next());
+    auto result = GlobalTabletSchemaMap::Instance()->emplace(metadata->schema());
```

Comment:
There maybe concurrent update?

Reply:
GlobalTabletSchemaMap::emplace is thread safe. When concurrency occurs, the same share ptr is returned. In the result variable, one inserted field is true and the other is false. These shared can be eliminated naturally with the cache.